evalml/objectives/standard_metrics.py

```diff
@@ -345,6 +345,17 @@ def objective_function(self, y_true, y_predicted, X=None):
         return metrics.mean_absolute_error(y_true, y_predicted)
 
 
+class MAPE(RegressionObjective):
```

Comment:
@freddyaboulton did we decide not to define `TimeseriesRegressionObjective`? Or did we simply not decide yet, haha, since this is the first objective explicitly defined for timeseries? I can't remember where we left that conversation.

Reply:
Previously, our objectives were only defined for a single problem type. We agreed to change this to a list of valid problem types and that our existing regression objectives should work for both regression and time series regression problems.

So long story short, we didn't define a `TimeSeriesRegressionObjective` because we didn't have any objectives that should only work for time series. We should make MAPE a `TimeSeriesRegressionObjective` in that case :) Good call.

Reply:
Ah got it, thanks @freddyaboulton 

So @ParthivNaresh , I think what we should do here is:
* Define `TimeseriesRegressionObjective` which subclasses `RegressionObjective` but allows only `ProblemTypes.TIME_SERIES_REGRESSION`
* Have `MAPE` subclass `TimeseriesRegressionObjective`

That sound good?

Reply:
@dsherry Sounds great!